python_ta/transforms/type_inference_visitor.py

```diff
@@ -536,15 +536,23 @@ def visit_classdef(self, node: astroid.ClassDef) -> None:
         # TODO: include node.instance_attrs as well?
         for attr in node.locals:
             attr_type = self.type_constraints.resolve(node.type_environment.lookup_in_env(attr)).getValue()
-            self.type_store.classes[node.name][attr].append(attr_type)
             if isinstance(attr_type, CallableMeta):
-                self.type_store.methods[attr].append(attr_type)
+                f_type = node.locals[attr][0].type
+                self.type_store.classes[node.name][attr].append((attr_type, f_type))
+                self.type_store.methods[attr].append((attr_type, f_type))
+            else:
+                self.type_store.classes[node.name][attr].append((attr_type, 'attribute'))
 
     ##############################################################################
     # Statements
     ##############################################################################
     def visit_attribute(self, node: astroid.Attribute) -> None:
         expr_type = node.expr.inf_type.getValue()
+
+        # case when expr_type is the type of a class
+        if hasattr(expr_type, '__name__') and expr_type.__name__ == 'Type':
```

Comment:
Seems like this happens too early. If we have an instance method `foo` in class`A`, `A.foo()` shouldn't autobind `self`, but you won't have a way of distinguishing this case if you make this update.

P.S. Would be good to add tests to check this.